internal/host/validator.go

```diff
@@ -271,114 +269,87 @@ type validator struct {
 	providerRegistry registry.ProviderRegistry
 }
 
-func (v *validator) isMediaConnected(c *validationContext) ValidationStatus {
-	return boolValue(c.host.MediaStatus == nil || *c.host.MediaStatus != models.HostMediaStatusDisconnected)
-}
-
-func (v *validator) isConnected(c *validationContext) ValidationStatus {
-	return boolValue(c.host.CheckedInAt.String() == "" || time.Since(time.Time(c.host.CheckedInAt)) <= v.hwValidatorCfg.MaxHostDisconnectionTime)
-}
-
-func (v *validator) printConnected(context *validationContext, status ValidationStatus) string {
+func (v *validator) isMediaConnected(c *validationContext) (ValidationStatus, string) {
+	status := boolValue(c.host.MediaStatus == nil || *c.host.MediaStatus != models.HostMediaStatusDisconnected)
 	switch status {
 	case ValidationSuccess:
-		return "Host is connected"
+		return status, "Media device is connected"
 	case ValidationFailure:
-		return "Host is disconnected"
+		return status, statusInfoMediaDisconnected
 	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return status, fmt.Sprintf("Unexpected status %s", status)
 	}
 }
 
-func (v *validator) printMediaConnected(context *validationContext, status ValidationStatus) string {
+func (v *validator) isConnected(c *validationContext) (ValidationStatus, string) {
+	status := boolValue(c.host.CheckedInAt.String() == "" || time.Since(time.Time(c.host.CheckedInAt)) <= v.hwValidatorCfg.MaxHostDisconnectionTime)
 	switch status {
 	case ValidationSuccess:
-		return "Media device is connected"
+		return status, "Host is connected"
 	case ValidationFailure:
-		return statusInfoMediaDisconnected
+		return status, "Host is disconnected"
 	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return status, fmt.Sprintf("Unexpected status %s", status)
 	}
 }
 
-func (v *validator) hasInventory(c *validationContext) ValidationStatus {
-	return boolValue(c.inventory != nil)
-}
-
-func (v *validator) printHasInventory(context *validationContext, status ValidationStatus) string {
+func (v *validator) hasInventory(c *validationContext) (ValidationStatus, string) {
+	status := boolValue(c.inventory != nil)
 	switch status {
 	case ValidationSuccess:
-		return "Valid inventory exists for the host"
+		return status, "Valid inventory exists for the host"
 	case ValidationFailure:
-		return "Inventory has not been received for the host"
+		return status, "Inventory has not been received for the host"
 	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return status, fmt.Sprintf("Unexpected status %s", status)
 	}
 }
 
-func (v *validator) hasMinCpuCores(c *validationContext) ValidationStatus {
-	if c.inventory == nil {
-		return ValidationPending
+func (v *validator) hasMinCpuCores(c *validationContext) (ValidationStatus, string) {
+	if c.inventory != nil {
```

Comment:
I think this would be better reversed. Generally people reading this expect the success case to be the least nested.